mininet/link.py

```diff
@@ -73,14 +77,24 @@ def setIP( self, ipstr, prefixLen=None ):
         # This is a sign that we should perhaps rethink our prefix
         # mechanism and/or the way we specify IP addresses
         if '/' in ipstr:
-            self.ip, self.prefixLen = ipstr.split( '/' )
-            return self.ifconfig( ipstr, 'up' )
+            ip, prefixLen = ipstr.split( '/' )
+            args = [ ipstr, 'up' ]
         else:
             if prefixLen is None:
                 raise Exception( 'No prefix length set for IP address %s'
                                  % ( ipstr, ) )
-            self.ip, self.prefixLen = ipstr, prefixLen
-            return self.ifconfig( '%s/%s' % ( ipstr, prefixLen ) )
+            ip, prefixLen = ipstr, prefixLen
+            args = [ '%s/%s' % ( ipstr, prefixLen ) ]
+
+        if ':' in ip:
+            if self.ip6 is not None:
+                self.ifconfig( 'del', '%s/%s' % ( self.ip6, self.prefixLen6 ) )
+            self.ip6, self.prefixLen6 = ip, prefixLen
+            args.insert( 0, 'add' )
+        else:
+            self.ip, self.prefixLen = ip, prefixLen
```

Comment:
Handle things differently only if the address was IPv6.  If it was IPv6, and there was previously an IPv6 address, then the old address has to be deleted before the new one can be added.

If the address was an IPv4 address, then there is no change.  It is backwards compatible with the old version.